omod/src/main/java/org/openmrs/module/pihcore/fragment/controller/dashboardwidgets/PregnancyIndicatorsFragmentController.java

```diff
@@ -0,0 +1,157 @@
+        }
+        if (patient == null) {
+            throw new IllegalArgumentException("No patient found. Please pass patient into the configuration");
+        }
+
+        model.put("app", app);
+        model.put("patient", patient);
+        String programUuid = PihCoreUtils.getConfigValue(app, "duringCurrentEnrollmentInProgram");
+
+        Map<String, SimpleObject> fields = new LinkedHashMap<>();
+
+        fields.put("status", SimpleObject.create(
+                "label", "pihcore.status",
+                "css", "red",
+                "obs", ui.message("pihcore.pregnant")
+        ));
+        Concept question = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_ESTIMATEDDUEDATE_UUID);
+        List<Obs> obsWithinProgram = null;
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, programUuid);
+        }
+
+        fields.put("dueDate", SimpleObject.create(
+                "label", "pihcore.delivery.dueDate",
+                "css", "red",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+
+        question = conceptService.getConceptByUuid(SierraLeoneConfigConstants.CONCEPT_PREGNANCYRISKFACTORS_UUID);
+        Concept answer = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_MULTIPLEBIRTH_UUID);
+        if ( question != null && answer != null) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), new HashSet<>(Arrays.asList(answer)), programUuid);
+        }
+        fields.put("multipleGestation", SimpleObject.create(
+                "label", "pihcore.multiple.birth",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? ui.message("general.yes") : ui.message("general.no")
+        ));
+
+        question = conceptService.getConceptByUuid(SierraLeoneConfigConstants.CONCEPT_DATETIMEOFBIRTH_UUID);
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("latestDelivery", SimpleObject.create(
+                "label", "pihcore.latest.delivery",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
+        ));
+        question = conceptService.getConceptByUuid(PihEmrConfigConstants.CONCEPT_TYPEOFDELIVERY_UUID);
+        if ( question != null ) {
+            obsWithinProgram = PihCoreUtils.getObsWithinProgram(patient, new HashSet<>(Arrays.asList(question)), null, null);
+        }
+        fields.put("deliveryType", SimpleObject.create(
+                "label", "pihcore.latest.delivery.type",
+                "obs", (obsWithinProgram!=null && obsWithinProgram.size() > 0 ) ? obsWithinProgram.get(0) : ""
```

Comment:
See my comment above.  Those this is a little trickier to figure out since you'd likely need to get the parent obs group. Maybe if there are more than one obs and they are different, just display both of them?

Reply:
@mogoodrich , I have updated the PR to display all different methods of delivery that are on the same Labour and Delivery form as the most recent Delivery date/time obs value:

<img width="936" alt="Screenshot 2024-11-22 at 6 26 02 AM" src="https://github.com/user-attachments/assets/cde3b6eb-c3ea-46a4-8512-b2c382cd7b83">